### DISCUSSION TITLE
RSECon 2024 connections

### DISCUSSION THREAD

**Discussion by metazool** [NERC-CEH/rse_group#40] [General]
General thread on connections we've made at RSECon 2024 that would be useful to follow up - for ourselves and for wider EDS

**Comment by metazool** [comment]
https://youtu.be/iB9NDFcU7wU?feature=shared - "Right to replicate" research infrastructure talk from Sarah at 2i2c - for thinking about DataLabs futures and partnerships, they specialise in managing and supporting _infrastructure-level_ JupyterHub / RStudio cloud for (networks of) research orgs, delegate data and applications

**Comment by metazool** [comment]
https://www.researchsoft.org/taskforces/ - Research Software Alliance task forces. They are starting a new infrastructure forum, scope maybe more cultural than cyberphysical, e.g. Zenodo, JOSS, aspects of Github. Speaker is Michelle

**Comment by metazool** [comment]
https://n8cir.org.uk/training-and-events/dri-retreat-2025/ - n8 CIR, 8 northern universities incl Lancaster collaborating on Computationally Intensive Research

DRI Retreat, 13th-17th Jan 2025 https://n8cir.org.uk/training-and-events/dri-retreat-2025/ - looking for role diversity in attendees for next session, who in FDRI / NC UK, more on infrastructure side, would be a good fit to go, sounds like mutual benefit

**Comment by metazool** [comment]
Becky Smith at James Hutton Inst, existing collaboration RETINA[sic] on agricultural soil moisture sensing, LoRaWAN, issues with proprietary kit installations, workshop invitees. Kit McLeod

**Comment by metazool** [comment]
https://yestem.org/tools/the-equity-compass/ - the "equity compass" (from keynote 1) - reminding me of "model the whole iceberg" / caucus scores    [link later] and FAIR/CARE for process and planning

**Comment by metazool** [comment]
https://www.humancellatlas.org/data-portal/#! and the amazing looking Web Atlas, [link later]

paper on RSE role from Horsfall et al, 2023

**Comment by metazool** [comment]
Anita Banerji (in panel session clashes with my talk), EIDA at Manchester, social science of RSE culture formation, remember to send references to work of Cat Hicks

**Comment by matthewcoole** [comment]
https://openghg.github.io/dashboard/ apparently are using zarr and xarray with some kind of object storage for remote gas sensor networks.

**Comment by metazool** [comment]
interesting poster on modified RaspPi camera modules for monitoring volcanic sulphur dioxide emissions in UV wavelengths
![Uploading 20240903_151748.jpg…]()


https://github.com/twVolc/PyCamPermanent

**Comment by metazool** [comment]
https://github.com/mlco2/codecarbon CodeCarbon for tracking software environmental cost (python)

**Comment by metazool** [comment]
https://cran.r-project.org/web/packages/orderly/vignettes/orderly.html - orderly for R, distributed graph for reproducible analysis, process traceability, like a next level evolution of {targets}, 

https://github.com/mrc-ide/PackIt for data distribution, "scales back down" from distributed versioned data to local running, metadata everywhere, i liked it

![20240903_161306](https://github.com/user-attachments/assets/38f6be01-d918-45b9-850c-e127cbebe503)

"stop making them draw the whole owl"

https://github.com/mrc-ide/pyorderly - very early days python port. an audience comment cites (unmaintained) `recipy`

**Comment by metazool** [comment]
https://mutmut.readthedocs.io/en/latest/index.html - mutation testing in a scenario where we are aiming for a very high level of test rigour (e.g. sensor data abstraction libraries for NC-UK)

**Comment by matthewcoole** [comment]
From Anne-Marie Imafidon's keynote yesterday, some people at UKCEH might be interested in [Stemettes](https://stemettes.org/).

The EEDI book club might also be interested in [Invisible Women](https://carolinecriadoperez.com/book/invisible-women/).

**Comment by matthewcoole** [comment]
Ann Gledson and Adrian Harwood from Manchester University gave an excellent overview of their flexible way of adopting parts of scrum for research projects, embracing some elements of scrum that work and make sense depending on the scale of the project and the engagement of researchers.

Interesting to see if using Github projects for managing the agile methodology and using 4 week sprints could work for us.

This is probably the kind of thing we'll really need to think about adopting in the coming weeks as we start getting more specific project time.

**Comment by metazool** [comment]
https://github.com/RSEToolkit - repos for the skills profile / frameworks and web application intended to share across group

useful both for input into internal discussion of "success profiles" and context for if we do any more engagement with the Turing led effort at #14

**Comment by metazool** [comment]
@phukeo tagged you here, but a lot of threads in this Discussions board cover our bootstrap meta-discussions, maybe useful for PML

**Comment by metazool** [comment]
https://elib.dlr.de/190851/ - FAIR4RS and Hermes, automation of FAIR research software publication, emphasis on metadata, packages as first class research artifact. interesting talk on the extent to which software quality criteria can be measured

https://github.com/hermes-hmc

**Comment by metazool** [comment]
https://calculator.green-algorithms.org/

Cambridge Dept of Public Health, consortium including SSI, tooling for workflows. questions about cost of data in storage, and scheduling compute load for when cleaner energy is more available

**Comment by metazool** [comment]
https://wellcome.org/who-we-are/positions-and-statements/environmental-sustainability-concordat - we are signatories to this Concordat. Presented jointly by Wellcome and UKRI from a perspective of funder requirements. How are resulting actions made easier and their outcomes measured? "Some elements of both carrot _and_ stick" - where do reproducibility incentives align and where do they diverge?

**Comment by metazool** [comment]
https://rosalindfranklininstitute.github.io/DisTRaX/index.html - Laura (and Laura!) from Rosalind Franklin Inst, a lot of good infrastructure work for a smaller org

https://github.com/rosalindfranklininstitute

Interesting answer in the q&a about the ethical consideration of environmental impact of research computing - casts another light on how we agree on the value of research questions at all!

Also "delete your data, please! we keep so much rubbish... is it cheaper to do the experiment again than to keep the data?", so many thoughtful awkward questions

**Comment by metazool** [comment]
https://www.nature.com/articles/s43588-023-00461-y "GREENER PRINCIPLES", how to make them specific to RSE practise / find where our role fits?

Governance
Responsibility
Estimation
Energy and embodied impact
New collab
Education
Research
SocRSE special interest group in formation, and greenresearchsoftware@jiscmail listserv

**Comment by phukeo** [comment]
Thank you Jo,

I really appreciate you passing these resources over to me!  Very helpful!

Stay in touch,

Ben

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by metazool** [comment]
@richfitz hope you don't mind being tagged in our group's RSECon mega-thread, not sure how to get in touch otherwise! (via @mikerspencer ? )

Had some follow-up questions about your [presentation on the orderly ecosystem](https://github.com/NERC-CEH/rse_group/discussions/40#discussioncomment-10533554) and your group's work on `pyorderly`.

Specifically I would appreciate 

1.  a copy of the slides to circulate to folks within our wider group and 
2.  an outline of the kind of use cases you're putting it to within MRC (if it's not covered in the slides, I missed the first half of the session :/)   

I'm thinking about it in the context of the [NC-UK](https://github.com/NERC-CEH/rse_group/discussions/28) five-year programme for sharing environmental monitoring data on a national scale. We've been asked to contribute proof-of-concept work to its early stages, in terms of your metaphor a rough but complete sketch of the owl! I'm not sure where your take on the `orderly` ecosystem fits, but reckon it might, thus wanting to know more of how RESIDE / MRC are using it in practise...

**Comment by richfitz** [comment]
* Slides are [here on google docs](https://docs.google.com/presentation/d/1Lh0b_CQhzpCCOtwFIxiRjzWl3_m_7nrMe1BSOxKc6mk/edit?usp=sharing)
* The talk will be available to you via [canapii](https://app.canapii.com/e/rsecon24/session/session-5fc3dcf6-396d-4ed5-b90a-dd05b2dea724) if you want to catch the first half. I think these won't be on YouTube for a few months though.
* I need to update my group's pages with information on our use cases, but briefly (and mostly in the pre-v2 version):
   * it is the backbone of the reporting framework for the [vaccine impact modelling consortium](https://vaccineimpact.org), where our researchers use it to analyse data and report back to our stakeholders - we use it to collect a centralised record of reports and as distribution mechanism
   * it was used throughout the UK covid response in our "pipeline" (in rather a looser sense than a bioinformatics sense as things were always changing) from dozens of data sets through a computationally intensive model through to simulations of future pandemic trajectories and reporting to the govt (figures on this are in the slides)
   * we use it to distribute cleaned/processed data within research groups (HIV and Malaria groups, though the use is independent of the diseases of course)

**Comment by mattjbr123** [comment]
Dumping my highlights here to neaten up and add to Joe's presentation later (I couldn't be bothered to get Quarto to work this evening xD)
Lots have already been picked up by my lovely RSE team colleagues.

- [Document checklist/assessment criteria](https://docs.google.com/document/d/1NuBSkRCY3wpLmuM0kGmZA8wPxTnqu4Ow6B-Ay1_vgoM/edit) and [worksheet](https://cehacuk.sharepoint.com/:x:/r/sites/ResearchSoftwareEngineeringCommunity/_layouts/15/Doc.aspx?sourcedoc=%7BCEF59DB7-CC17-4FE7-8B13-2104DBD33D3D%7D&file=Reproducibility_Assessment_Worksheet.xlsx&wdOrigin=TEAMS-MAGLEV.p2p_ns.rwc&action=default&mobileredirect=true) for where organisations are for software reproducibility
- [A name change policy working group](https://ncpwg.org/resources/authors/) aiming to make it easier for researchers to change their names and not be harmed by it, given academia's focus on your publishing/contribution record
- [RO-Crate](https://www.researchobject.org/ro-crate/). A way of packaging metadata and provenance with data. Could be useful for FDRI. Follow up with a [previous assessment](https://wiki.ceh.ac.uk/display/ad/ro-crate) of it conducted in the EDS team. [Orderly](https://github.com/mrc-ide/pyorderly) is in a similar space too. 
- Some [performance and profiling training resources](https://rse.shef.ac.uk/pando-python/index.html)
- Some [testing training resources](https://github.com/abhidg/advanced-python-testing)
- Some [FAIR4RS (FAIR for Research Software) training resources](https://carpentries-incubator.github.io/fair-research-software/00-introduction.html)
- Work at NOC that has close parallels with what I am trying to do in FDRI. [Presentation](https://docs.google.com/presentation/d/18RZSS0LEIYOHK0Qhh_UWaCh00lhfXlfO/edit#slide=id.p1) here, [streamlit-based visualisation app](https://github.com/NOC-OI/class_streamlit_zarr) here, cute little command line application aiming to simplify the netcdf -> rechunk -> zarr pipeline [here](https://github.com/NOC-OI/msm-os).
